docs/architecture/adr-005.md

```diff
@@ -0,0 +1,61 @@
+This setup is potentially suited for scenarios where the consumer blockchains contain a more experimental code, because any potential issues will not
+directly affect their provider blockchain. For example, if the consumer blockchain panics and halts due to a bug, the provider blockchain does not halt and continues operating.
+
+Gravity Bridge is an experimental module (plus its surrounding infrastructure) that allows bi-directional ERC20 token transfers with Ethereum and is included under an experimental flag in Cronos (see [ADR 001](./adr-001.md)), i.e. it is disabled in the normal operation.
+
+
+## Decision
+
+Gravity Bridge will be separated out to a dedicated network that will have the consumer-provider relationship with Cronos.
+The implementation can be done in the following stages:
+
+1. A new blockchain source tree is created for the Gravity Bridge-related code:
+- This can either be in a dedicated repository or inside the Cronos repository using the Go 1.18 workspace feature (TODO: which one?).
+- The new codebase will only import needed Cosmos SDK modules, ibc-go, and Gravity Bridge.
+- It could potentially also import the packet forwarder middleware: https://github.com/strangelove-ventures/packet-forward-middleware
+- Alternatively, it could implement something similar to the packet forwarder middleware in order to automate incoming token transfers (i.e. so that incoming transfers from Ethereum are automatically sent via IBC to Cronos).
+
+2. The Gravity Bridge-related code is removed from the Cronos module/application.
+
+3. The x/ccv module is added to the Cronos and the Gravity Bridge applications.
+
+4. The public contracts and standards may be revised to remove the `send_to_ethereum` function and related events.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- Isolation of Gravity Bridge: any potential issues discovered in it will not affect the overall Cronos network's operation.
+- Independent processing of transactions: any transaction spikes in the EVM or Gravity Bridge modules will not affect each other directly.
+- Storage sharding: the storage needed by Gravity Bridge can live on a different node from the one executing Cronos with the EVM module.
+- Simplified interface for cross-chain token transfers: only IBC-related logic needs to be considered inside EVM.
+- Independent upgrades: any breaking changes in Gravity Bridge will not demand upgrades on Cronos (and vice versa).
+
+### Negative
+
+- Complexity and maintenance overhead of a separate network codebase and operation. (Note that the Gravity Bridge infrastructure demands an inherent operational complexity, e.g. the need to run Ethereum full nodes, regardless of where the module is included.)
```

Comment:
I think that summarise the pro and cons. 

While technically all "macro" features in Cronos could be created into a separate chain. We need to evaluate if there the gravity bridge as a standalone chain can have its own roadmap (future enhancement, new features etc) and thus could justify the operation costs (server, maintenance, monitoring, data storage cost)

Reply:
https://github.com/Gravity-Bridge/Gravity-Bridge/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement

I assume one future breaking change will be ERC721 and ERC1155 transfers?

Reply:
+ multi chain support maybe